scripts/license/add_license_header_test.go

```diff
@@ -0,0 +1,201 @@
+				   This file is part of some old version
+				   of the Sonic Client`
+	_, err = tmpFile.WriteString(addPrefix(oldLicense, "//") + "\n\npackage main\n\nfunc main() {}\n")
+	require.NoError(t, err)
+	require.NoError(t, tmpFile.Close())
+
+	license, err := getLicenseHeader()
+	require.NoError(t, err)
+	err = processFiles(tmpDir, ".go", "//", string(license), false, false)
+	require.NoError(t, err)
+
+	content, err := os.ReadFile(tmpFile.Name())
+	require.NoError(t, err)
+	require.Contains(t, string(content), addPrefix(string(license), "//"))
+
+	require.NotContains(t, string(content), oldLicense)
+}
+
+func Test_Adds_LicenseHeader(t *testing.T) {
+	// make a temporary folder
+	tmpDir := t.TempDir()
+	// make a temporary file in that folder
+	tmpFile, err := os.Create(tmpDir + "/test_license.go")
+	require.NoError(t, err)
+	defer func() { _ = tmpFile.Close() }()
+
+	license, err := getLicenseHeader()
+	require.NoError(t, err)
+
+	// write a sample license header to the file
+	_, err = tmpFile.WriteString("package main\n\nfunc main() {}\n")
+	require.NoError(t, err)
+	require.NoError(t, tmpFile.Close())
+
+	err = processFiles(tmpDir, ".go", "//", license, false, false)
+	require.NoError(t, err)
+
+	content, err := os.ReadFile(tmpFile.Name())
+	require.NoError(t, err)
+	extendLicenseHeader := addPrefix(license, "//")
+	require.Contains(t, string(content), extendLicenseHeader)
+}
+
+func copyFile(src, dst string) error {
+	originalContent, err := os.ReadFile(src)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(dst, originalContent, 0644)
+}
+
+func Test_Detects_DoubleHeader(t *testing.T) {
+	// make a temporary folder
+	tmpDir := t.TempDir()
+	// make a temporary file in that folder
+	tmpFile, err := os.Create(tmpDir + "/test_license.go")
+	require.NoError(t, err)
+	defer func() { _ = tmpFile.Close() }()
+
+	license, err := getLicenseHeader()
+	require.NoError(t, err)
+
+	// write a sample license header to the file
+	_, err = tmpFile.WriteString(addPrefix(string(license), "//") + addPrefix(string(license), "//") + "\npackage main")
+	require.NoError(t, err)
+
+	// Check for double license headers
+	require.Error(t, checkDoubleHeader(tmpFile.Name(), "//"))
+}
+
+func Test_OnlyOnEmptyLineAfterHeader(t *testing.T) {
+	// make a temporary folder
+	tmpDir := t.TempDir()
+	// make a temporary file in that folder
+	tmpFile, err := os.Create(tmpDir + "/test_license.go")
+	require.NoError(t, err)
+	defer func() { _ = tmpFile.Close() }()
+
+	license, err := getLicenseHeader()
+	require.NoError(t, err)
+
+	// write a sample license header to the file
+	_, err = tmpFile.WriteString(addPrefix(string(license), "//") + "\npackage main\n\nfunc main() {}\n")
+	require.NoError(t, err)
+
+	// Check for double license headers
+	err = processFiles(tmpDir, ".go", "//", license, false, false)
+	require.NoError(t, err)
+
+	content, err := os.ReadFile(tmpFile.Name())
+	require.NoError(t, err)
+	firstEmptyLine := false
+	for _, line := range string(content) {
```

Comment:
I think here you are iterating through characters, not lines ...

Reply:
https://github.com/0xsoniclabs/sonic/pull/335/commits/ea236981378815b28b94fb3717a15fd8ace6fb49